pytket/extensions/cutensornet/mps/simulation.py

```diff
@@ -73,7 +73,8 @@ def simulate(circuit: Circuit, algorithm: ContractionAlg, **kwargs: Any) -> MPS:
     sorted_gates = _get_sorted_gates(circuit)
 
     # Apply the gates
-    with mps.init_cutensornet():
+    with CuTensorNetHandle() as libhandle:

```

Comment:
I wonder if we could say pass MPS objects to `CuTensorNetHandle` and call their `.set_libhandle()` from there (i.e. less boilerplate)?

Reply:
Could be done, but I think the opposite approach is more desirable and will cover all use cases; so it's simpler API if we don't add this option.

Reply:
Agreed! :)